CMakeLists.txt

```diff
@@ -10,39 +10,21 @@ if (APPLE)
     set(CMAKE_NO_SYSTEM_FROM_IMPORTED TRUE)
 endif()
 
-set(SUPPORTED_PLATFORMS "Win32" "UWP" "Android" "Apple")
-if (ANDROID)
-    set(BABYLON_NATIVE_PLATFORM "Android" CACHE STRING "Target platform for Babylon Native.")
-elseif (APPLE)
-    set(BABYLON_NATIVE_PLATFORM "Apple" CACHE STRING "Target platform for Babylon Native.")
+if(ANDROID)
+    set(BABYLON_NATIVE_PLATFORM "Android")
+elseif(APPLE)
+    set(BABYLON_NATIVE_PLATFORM "Apple")
+elseif(WINDOWS_STORE)
+    set(BABYLON_NATIVE_PLATFORM "UWP")
+elseif(WIN32)
+    set(BABYLON_NATIVE_PLATFORM "Win32")
 else()
-    set(BABYLON_NATIVE_PLATFORM "Win32" CACHE STRING "Target platform for Babylon Native.")
-endif()
-
-if(NOT(BABYLON_NATIVE_PLATFORM IN_LIST SUPPORTED_PLATFORMS))
-    message(FATAL_ERROR "Unrecognized platform: ${BABYLON_NATIVE_PLATFORM}")
-endif()
-
-if(BABYLON_NATIVE_PLATFORM STREQUAL "UWP" AND NOT(WINDOWS_STORE))
-    message(FATAL_ERROR "\nUWP builds must target Windows Store. Make sure to designate\n\t\"-D CMAKE_SYSTEM_NAME=WindowsStore -D CMAKE_SYSTEM_VERSION=10.0\"\nwhen configuring for UWP.")
+    message(FATAL_ERROR "Unrecognized platform: ${CMAKE_SYSTEM_NAME}")
```

Comment:
Is `${CMAKE_SYSTEM_NAME}` defined all the time?

Reply:
I don't know of any cases where it wouldn't be defined. From [this doc](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SYSTEM_NAME.html), it looks like it's guaranteed to have a default definition at the very least for Windows, macOS, and Linux systems.